internal/servers/destinations.go

```diff
@@ -87,16 +87,15 @@ func (s *DestinationServer) Write2(msg pb.Destination_Write2Server) error {
 	for {
 		r, err := msg.Recv()
 		if err != nil {
+			close(resources)
 			if err == io.EOF {
-				close(resources)
 				if err := eg.Wait(); err != nil {
-					return fmt.Errorf("got EOF. failed to wait for plugin: %w", err)
+					return fmt.Errorf("got EOF. plugin returned: %w", err)
```

Comment:
`EOF` is not an error necessarily (it usually means the stream was closed), so I'm not sure we want to report it. Also to be consistent we should add `plugin returned` to everywhere we do `eg.Wait()`, like lines `105` and `114` in this file.

You can see https://github.com/cloudquery/plugin-sdk/pull/461/files for reference.

We can also reduce the nesting of handling `EOF`, but that can be done in another PR, see https://github.com/cloudquery/plugin-sdk/pull/463 for reference